python/lsst/pipe/base/graph/graph.py

```diff
@@ -175,6 +176,14 @@ def _buildGraphs(
         and the graph that holds the relations between quanta
         """
         self._metadata = metadata
+
+        # Save packages to metadata
+        metadata2 = {}
+        if self._metadata is not None:
+            metadata2 = dict(self._metadata)
+        metadata2["packages"] = Packages.fromSystem()
+        self._metadata = metadata2
```

Comment:
I think what you wrote is fine, but if you are now going to always be adding to the metadata now may I suggest:
```
self._metadata = dict(metadata) if metadata is not None else {}
Self._metadata[‘packages’] = Packages.fromSystem()
```
Fewer round trips and copies being made.